ESLint-Formatter.py

```diff
@@ -143,11 +143,64 @@ def get_node_path():
     return node
 
   @staticmethod
+  def get_config_path():
+    cwd = os.getcwd()
+    folders = cwd.split('/')
+    return PluginUtils.get_eslint_config(cwd, folders)
+
+  @staticmethod
+  def get_eslint_config(root, folders):
+    cfs = ['.eslintrc.js', '.eslintrc.yaml', '.eslintrc.yml', '.eslintrc.json', '.eslintrc']
+    dir = None
+    for cf in cfs:
+      fd = os.path.join(root, cf)
+      if os.path.isfile(fd):
+        dir = fd
+        return dir
+
+    if dir == None:
+      f = folders.pop()
+      if f:
+        d = root.split(f)[0]
+        if d:
+          dir = PluginUtils.get_eslint_config(d, folders)
+          return dir
+
+    return False
+
+  @staticmethod
+  def get_eslint(root, folders):
+    # get from project's node_modules
+    ePath = 'node_modules/.bin/eslint'
+    eslint = os.path.join(root, ePath)
+
+    if os.path.isfile(eslint):
+      return eslint
+    else:
+      fp = folders.pop()
+      if fp:
+        fd = root.split(fp)[0]
+        if fd:
+          eslint = PluginUtils.get_eslint(fd, folders)
+          return eslint
+
+    return False
+
+
+  @staticmethod
   def get_eslint_path():
     platform = sublime.platform()
-    eslint = PluginUtils.get_pref("eslint_path").get(platform)
-    print("Using eslint path on '" + platform + "': " + eslint)
-    return eslint
+    cwd = os.getcwd()
```

Comment:
What is the current working directory when running a sublime plugin? Is it the folder of the file that currently has focus? Is it the top level folder of the project? Is it some sublime application folder?